src/pybullet_helpers/geometry.py

```diff
@@ -0,0 +1,82 @@
+"""Pybullet helper class for geometry utilities."""
+from typing import NamedTuple, Sequence
+
+import numpy as np
+import pybullet as p
+from pybullet_utils.transformations import euler_from_quaternion, \
+    quaternion_from_euler
+
+from predicators.src.structs import Array, Pose3D, Quaternion, RollPitchYaw
+
+
+class Pose(NamedTuple):
+    """Pose which is a position (translation) and rotation.
+
+    We use a NamedTuple as it supports retrieving by integer indexing
+    and most closely follows the pybullet API.
+    """
+
+    position: Pose3D
+    quat_xyzw: Quaternion = (0.0, 0.0, 0.0, 1.0)
+
+    @classmethod
+    def from_rpy(cls, translation: Pose3D, rpy: RollPitchYaw) -> "Pose":
+        """Create a Pose from translation and Euler roll-pitch-yaw angles."""
+        return cls(translation, quaternion_from_euler(*rpy))
+
+    @property
+    def orientation(self) -> Quaternion:
+        """The default quaternion representation is xyzw as followed by
+        pybullet."""
+        return self.quat_xyzw
+
+    @property
+    def quat_wxyz(self) -> Quaternion:
+        """Get the wxyz quaternion representation."""
+        return (
+            self.quat_xyzw[3],
```

Comment:
what's with this transposition?

Reply:
Just a different quaternion ordering. The default in pybullet is xyzw so I think we can remove this for now and re-add it when needed.